### DISCUSSION TITLE
issue:  Appending and removing nested fields is not working properly

### DISCUSSION THREAD

**Discussion by john45321** [react-hook-form/react-hook-form#7936] [Q&A]
### Version Number

7.27.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-nested-field-array-881j8b

### Steps to reproduce

**Problem 1:**

1.  Go to the codesandbox link
2. Enter value `2` inside the `totalDays` input
3. Click on the append button 2 times to dynamically create 2 sections
4. After this, click on the 2nd day box and again click on the 1st day box
5. Finally click on the append button
6. Notice: It will remove the last section instead of adding a new one

**Problem 2:**

1.  Go to the codesandbox link
2. Enter value `2` inside the `totalDays` input
3. Click on append button 2 times to dynamically create 2 sections
4. After this, click on the 2nd day box and again click on the 1st day box
5. Finally click on the remove button
6. Notice: It will remove the last two sections instead of removing at given index

### Expected behaviour

**Expected behavior on the problem 1:**

When the user clicks on the append button on step 5, It should add a new section instead of removing the last section. 

![sample-1](https://user-images.githubusercontent.com/59892334/155878866-30ce05b2-b4d0-4659-8b1c-c0412d40f29c.gif)

**Expected behavior on the problem 2:**

When the user clicks on the remove button on step 5, It will remove the last two sections instead of removing at given index.

![sample-22](https://user-images.githubusercontent.com/59892334/155879137-17a69bfe-26ec-46bc-a1b8-be3e0b1fb774.gif)

Example of the form state structure:
```js
{
  totalDays: '2',
  days: [
    {
      selected: true,
      error: false,
      slots: [
        {
          name: '',
          capacity: 0
        },
        {
          name: '',
          capacity: 0
        },

        {
          name: '',
          capacity: 0
        }
      ]
    },
    {
      selected: false,
      error: false,
      slots: [
        {
          name: '',
          capacity: 0
        }
      ]
    }
  ]
}
```

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
can you please send a codesandbox that is isolated to this issue rather than a whole project?

especially with logic as such:
<img width="854" alt="Screen Shot 2022-02-27 at 9 55 22 pm" src="https://user-images.githubusercontent.com/10513364/155879542-f64b3f6e-715e-4a2d-8067-21459b8607c6.png">
